dashboard/src/app/(authenticated)/components/tables/WfSpecTable.tsx

```diff
@@ -1,30 +1,57 @@
+import { FC, Fragment, useMemo } from 'react'
+import { SearchResultProps } from '.'
 import { WfSpecId } from 'littlehorse-client/proto'
+
 import { TagIcon } from 'lucide-react'
-import Link from 'next/link'
-import { FC, Fragment } from 'react'
-import { SearchResultProps } from '.'
+import { Separator } from '@/components/ui/separator'
+import { useRouter } from 'next/navigation'
 
 export const WfSpecTable: FC<SearchResultProps> = ({ pages = [] }) => {
+  const router = useRouter()
+
   if (pages.length === 0) {
     return <div className="flex min-h-[360px] items-center justify-center text-center italic">No WfSpecs</div>
   }
 
+  const wfSpecs = useMemo(() => {
+    const specMap = new Map<string, wfSpecData>()
+
+    pages
+      .flatMap(page => page.results)
+      .reverse()
+      .forEach(({ name, majorVersion, revision }: WfSpecId) => {
+        if (!specMap.has(name)) {
+          specMap.set(name, { name, latestVersion: `${majorVersion}.${revision}` })
+        }
+      })
+
+    return Array.from(specMap.values())
+  }, [pages])
+
   return (
     <div className="py-4">
-      {pages.map((page, i) => (
-        <Fragment key={i}>
-          {page.results.map(({ name, majorVersion, revision }: WfSpecId) => (
-            <div key={`${name}.${majorVersion}.${revision}`} className="my-2 flex gap-2">
-              <Link className="underline hover:no-underline" href={`/wfSpec/${name}/${majorVersion}.${revision}`}>
-                {name}
-              </Link>
+      <div className="flex max-h-[600px] flex-col overflow-auto">
+        {wfSpecs.map(wfSpec => (
+          <Fragment key={wfSpec.name}>
```

Comment:
Good job using the `wfSpec.name` as the key instead of the `index` I believe we should use `wfSpec.id` though because there might be multiple `WfSpec`s with the same name

Is what I was going to say but I realize that that `wfSpec.name` is unique as the map is pointed to the latest version. So good job!